examples/eth_bots/execute_agent_trades.py

```diff
@@ -40,89 +119,126 @@ def execute_agent_trades(
             )
             # do_trade
             trade_amount: int = trade_object.trade.trade_amount.scaled_value
-            maturity_time = (
-                trade_object.trade.mint_time
-                + hyperdrive_market.position_duration.years * elftime.TimeUnit.SECONDS.value
-            )
+            position_duration_years = hyperdrive_market.position_duration.years * elftime.TimeUnit.SECONDS.value
+            mint_time: FixedPoint = trade_object.trade.mint_time
+            maturity_time: int = (mint_time + position_duration_years).scaled_value
             # TODO: The following variables are hard coded for now, but should be specified in the trade spec
+            # HyperdriveMarketAction does have min_amount_out
             max_deposit = trade_amount
             min_output = trade_amount + 1
             min_apr = int(1)
             max_apr = int(1e18)
             as_underlying = True
             # sort through the trades
-            # TODO: raise issue on failure by looking at `tx_receipt` returned from function
+            # TODO: figure out fees paid
             if trade_object.trade.action_type == MarketActionType.OPEN_LONG:
-                _ = eth.smart_contract_transact(
+                trade_result = transact_and_parse_logs(
                     web3,
                     hyperdrive_contract,
                     account,
                     "openLong",
-                    trade_amount,  # base amount
-                    min_output,
-                    account.checksum_address,
-                    as_underlying,
+                    *(trade_amount, min_output, account.checksum_address, as_underlying),
+                )
+                mint_time = trade_result.maturity_time - position_duration_years
+                wallet_deltas = WalletDeltas(
+                    balance=Quantity(
+                        amount=-trade_result.base_amount,
+                        unit=TokenType.BASE,
+                    ),
+                    longs={mint_time: Long(trade_result.bond_amount)},
                 )
             elif trade_object.trade.action_type == MarketActionType.CLOSE_LONG:
-                _ = eth.smart_contract_transact(
+                trade_result = transact_and_parse_logs(
                     web3,
                     hyperdrive_contract,
                     account,
                     "closeLong",
-                    maturity_time,
-                    trade_amount,  # base amount
-                    min_output,
-                    account.checksum_address,
-                    as_underlying,
+                    *(maturity_time, trade_amount, min_output, account.checksum_address, as_underlying),
```

Comment:
what's this doing?

Reply:
```
                trade_result = transact_and_parse_logs(
                    web3,
                    hyperdrive_contract,
                    account,
                    "openLong",
                    *(trade_amount, min_output, account.checksum_address, as_underlying),
                )
```
is equivalent to 
```
                trade_result = transact_and_parse_logs(
                    web3,
                    hyperdrive_contract,
                    account,
                    "openLong",
                    trade_amount,
                    min_output,
                    account.checksum_address,
                    as_underlying,
                )
```

I just thought the former was more clear, bc everything in `()` is the arguments to the smart contract function.

I had also at one point done this:
```
               fn_args = (trade_amount, min_output, account.checksum_address, as_underlying)
               trade_result = transact_and_parse_logs(
                    web3,
                    hyperdrive_contract,
                    account,
                    "openLong",
                    *fn_args,
                )
```

I am indifferent, whatever you prefer is good with me.